x/campaign/types/message_burn_vouchers.go

```diff
@@ -0,0 +1,54 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+var _ sdk.Msg = &MsgBurnVouchers{}
+
+func NewMsgBurnVouchers(creator string, campaignID uint64, vouchers sdk.Coins) *MsgBurnVouchers {
+	return &MsgBurnVouchers{
+		Creator:    creator,
+		CampaignID: campaignID,
+		Vouchers:   vouchers,
+	}
+}
+
+func (msg *MsgBurnVouchers) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgBurnVouchers) Type() string {
+	return "BurnVouchers"
+}
+
+func (msg *MsgBurnVouchers) GetSigners() []sdk.AccAddress {
+	creator, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{creator}
+}
+
+func (msg *MsgBurnVouchers) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgBurnVouchers) ValidateBasic() error {
```

Comment:
Since we have the campaign ID in the message and the campaign ID in voucher's denom, we should statically check that all vouchers in the messages are linked to the provided campaign